automod_legacy/rules.go

```diff
@@ -215,6 +227,34 @@ OUTER:
 
 		checked = append(checked, id)
 
+		InvitesCache.RLock()
+		if len(InvitesCache.CacheMap) == 0 {
+			go inviteCacheGC()
+		}
+
+		for guild, cache := range InvitesCache.CacheMap {
+			if guild == guildID {
+				for _, cachedInvite := range cache {
+					if cachedInvite.Invite == id {
+						// Ignore invites to this server
+						InvitesCache.RUnlock()
+						continue OUTER
+					}
+				}
+			} else {
+				for _, cachedInvite := range cache {
+					if cachedInvite.Invite == id {
+						// If the invite is present on our cache, we return true even if it is not valid anymore
+						// This is to prevent making API Calls about invites which have a restrict rate limit
+						// Because otherwise we would most likely hit those limits
+						InvitesCache.RUnlock()
+						return true
+					}
+				}
+			}
```

Comment:
```suggestion
			for _, cachedInvite := range cache {
				if cachedInvite.Invite == id {
					InvitesCache.RUnlock()
					if guild == guildID {
						// Ignore invites to this server
						continue OUTER
					}
					// If the invite is present on our cache, we return true even if it is not valid anymore
					// This is to prevent making API Calls about invites which have a restrict rate limit
					// Because otherwise we would most likely hit those limits
					return true
				}
			}
```
This looks to be equivalent? Bit easier to understand for me as well but dunno

Reply:
Your are right!